### ISSUE TITLE
Observable source assets don't work if a partitioned asset exists in the project

### ISSUE THREAD

**Issue by maxfirman** [dagster-io/dagster#13516]
### Dagster version

1.2.6

### What's the issue?

If a project contains both a partitioned asset and an observable source asset, clicking the "Observe sources" button for the source asset will cause an error popup with the following message:

`__ASSET_JOB cannot be executed with the provided config.`

Removing the partitioned asset will resolve the issue, i.e the "Observe sources" button will trigger the observation function run as expected.

### What did you expect to happen?

The presence of a partitioned asset should not cause the Observe sources button to throw an error.

### How to reproduce?

Scaffold a new Dagster project with:

`dagster project scaffold --name <project-name> `

Add the following code to the `assets.py` file:

```python
from dagster import (
    DataVersion,
    observable_source_asset,
    DailyPartitionsDefinition,
    asset,
)


@observable_source_asset
def foo_source_asset():
    return DataVersion("foo")


@asset(partitions_def=DailyPartitionsDefinition(start_date="2023-03-01"))
def bar_asset():
    return "bar"

```

Run Dagit and click on the "Observe sources" button to see the error.

Removing `bar_asset` should resolve the error, and allow the observation job to run.

### Deployment type

Dagster Helm chart

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by tacastillo** [comment]
Hi Max! Thank you for reporting this!

@sryza are you the right person to give a heads-up on this?

**Comment by sryza** [comment]
This is relevant to what @smackesey is currently working on. Sean - mind chiming in here?

**Comment by smackesey** [comment]
This is fixed (#13482) in the release that will go out tmrw.
